src/main/java/org/opentestsystem/rdw/reporting/service/impl/ExamItemServiceImpl.java

```
@@ -184,61 +185,65 @@ private boolean itemFormatIsValidToHaveAnswerKey(XPath xPath, Document xmlDoc) {
 
-        return valueNode.getNodeType() == Node.ATTRIBUTE_NODE
-                ? valueNode.getNodeValue()
-                : valueNode.getTextContent();
+            return valueNode.getNodeType() == Node.ATTRIBUTE_NODE
+                    ? valueNode.getNodeValue()
+                    : valueNode.getTextContent();
+        } catch (XPathExpressionException e) {
+            // This exception is already pretty well guarded against since we are checking if
+            // the node exists before going to get the value.
+
+            logger.error("Error parsing xml expression: " + expression, e);
+            return "";
+        }
     }
 
-    private String encodeImages(String html, String id) throws IOException {
+    private String encodeImages(String html, String itemNumber) {
         String imgRegex = "<img[^>]+src\\s*=\\s*['\"]([^'\"]+)['\"][^>]*>";
         String result = html;
 
         Matcher matcher = Pattern.compile(imgRegex).matcher(html);
 
         while (matcher.find()) {
             String imageFileName = matcher.group(1);
-            InputStream imageStream = artifactRepository.getArtifactFromItemDirectory(id, imageFileName);
-            String encoding = Base64
-                    .getEncoder()
-                    .encodeToString(ByteStreams.toByteArray(imageStream));
+            try {
+                InputStream imageStream = artifactRepository.getArtifactFromItemDirectory(itemNumber, imageFileName);
+                String encoding = Base64
+                        .getEncoder()
+                        .encodeToString(ByteStreams.toByteArray(imageStream));
+
+                String mimeType = URLConnection.guessContentTypeFromName(imageFileName);
+                result = result.replace(imageFileName, "data:" + mimeType + ";base64," + encoding);
+            } catch (IOException e) {
+                String message = MessageFormat.format("Unable to find referenced image({0}) for item number: {1}", imageFileName, itemNumber);
```

It is okay to catch and eat this exception with nothing more than a log message?
I believe this is a WARN level, not ERROR. The app will continue to run without any compromise in behavior.

I suggest that, when catching an exception like this, you do *not* want to emit the entire stack trace. And you can just inline this: logger knows how to format messages. The result would be something like:
```
    logger.warn("Unable to find referenced image({}) for item number {}: {}", imageFileName, itemNumber, e.getMessage());
```